src/index.js

```diff
@@ -1,26 +1,34 @@
 // @flow
 import React from "react";
 
+declare function __webpack_require__(pathOrId: string | number): any;
+declare var __webpack_modules__: {};
 type GenericComponent<Props> = Class<React.Component<{}, Props, mixed>>;
 type LoadedComponent<Props> = GenericComponent<Props>;
 type LoadingComponent = GenericComponent<{}>;
 
+let isWebpack = typeof __webpack_require__ !== "undefined";
+let requireFn = isWebpack ? __webpack_require__ : module.require;
+
 let babelInterop = obj => {
   // $FlowIgnore
   return obj && obj.__esModule ? obj.default : obj;
 };
 
 let tryRequire = (pathOrId: string | number) => {
   try {
+    if (isWebpack && !__webpack_modules__[pathOrId]) {
+      // if it's not in webpack modules, we wont be able
+      // to load it. If we attempt to, we mess up webpack's
+      // internal state, so we exit early
+      return null;
```

Comment:
Here's a part of `__webpack_require__`:

```javascript

/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {

/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId])
/******/ 			return installedModules[moduleId].exports;

/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};

/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
// ... yada yada
```

So what's happening here is if the given `moduleId` exists in `installedModules`, it exits early with that module's `exports`. If it doesn't, then an empty one is created and then the module function is called with `.call`. However, in our case, the module has yet to be loaded so the `.call` call throws an error. We catch and ignore that error which is fine, but then the next time we come around (when loading it asynchronously) we're going to exit early with that empty `exports` object that it was initialized with.

So to fix this, we check whether `__webpack_modules__` (which I think is just the same thing as `installedModules`) already has the module. If it doesn't, then we're going to skip calling `__webpack_require__` because it's going to fail anyway (and mess up the state of `installedModules`).

Because this is kinda confusing I thought it'd be good to include a comment explaining this briefly.